### ISSUE TITLE
Fix setup.py

### ISSUE THREAD

**Issue by rtnpro** [projectatomic/atomic#112]
Issues:
- Importing ``__version__`` from Atomic package causes a race condition, where a new user cannot access ``setup.py`` without pre installing dependencies.
- ``python setup.py develop`` does not work.
- Install Python dependencies on running ``setup.py {develop,install}``
- Fix linting errors

**Comment by rhatdan** [comment]
Not sure how this is supposed to work.  

You want to just grab setup.py without having Atomic/atomic.py in the same directory?
What is 
python setup.py develop 
do?

```
 python setup.py develop
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: invalid command 'develop'
```

**Comment by rtnpro** [comment]
@rhatdan 

``python setup.py develop``
allows one to installs the package (typically just a source folder) in a way that allows you to conveniently edit your code after its installed to the (virtual) environment and have the changes take effect immediately. (w.r.t. http://stackoverflow.com/a/19048754)
http://pythonhosted.org//setuptools/setuptools.html#develop-deploy-the-project-source-in-development-mode

**Comment by rhatdan** [comment]
@rtnpro Thanks.   I figured out what you were talking about when I saw your pull requests.

**Comment by rtnpro** [comment]
@rhatdan :)

**Comment by rtnpro** [comment]
I am trying to setup this repo from scratch again, today, and I ran into this error:
```
(atomic)[rtnpro@rtnpro atomic]$ python setup.py develop
Traceback (most recent call last):
  File "setup.py", line 9, in <module>
    from Atomic import __version__
  File "/home/rtnpro/workspace/projectatomic/atomic/Atomic/__init__.py", line 3, in <module>
    from .pulp import PulpServer, PulpConfig
  File "/home/rtnpro/workspace/projectatomic/atomic/Atomic/pulp.py", line 2, in <module>
    import requests
ImportError: No module named requests
(atomic)[rtnpro@rtnpro atomic]$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 9, in <module>
    from Atomic import __version__
  File "/home/rtnpro/workspace/projectatomic/atomic/Atomic/__init__.py", line 3, in <module>
    from .pulp import PulpServer, PulpConfig
  File "/home/rtnpro/workspace/projectatomic/atomic/Atomic/pulp.py", line 2, in <module>
    import requests
ImportError: No module named requests
```
It seems that this bug got introduced by 106878815f5a5dd61b0e527c502fb6eeaf0c9c67 

I do not see a way to execute
``from Atomic import __version__``
without installing requests beforehand. The fix in this pull request to fetch version from a common place seemed to work fine.

Please share your thoughts.

**Comment by cgwalters** [comment]
Why not install the requests package on the system?  E.g. `$pkg install python-requests`?

**Comment by cgwalters** [comment]
Is there a way we can make `python setup.py develop` not require `pkg_resources` at runtime?  If not, I think we should stop caring about `python setup.py develop`.  I can see it being handy for websites or whatever, but `/usr/bin/atomic` should be more integrated with the OS.  (Among other things it needs to run as root right now...)

**Comment by ashcrow** [comment]
@rtnpro is this still an issue after #180?

**Comment by cdrage** [comment]
timeouttttt! pinging @rtnpro again, I'll try to catch him on irc :)

**Comment by rtnpro** [comment]
@cdrage @ashcrow It should still be a problem because 

``from Atomic import __version__``

will fail. The reason being the imports in https://github.com/cgwalters/atom/blob/6f94706621dce8ccff110febd36639e91dd35f8b/Atomic/__init__.py will try to access the yet uninstalled dependencies. For example, ``pulp`` modules will try to look for ``requests``.

**Comment by rhatdan** [comment]
Do we still have this issue, or can I close this?

**Comment by ashcrow** [comment]
The original error doesn't seem to be, but attempting to import from the repo still doesn't work due to a missing library not in ``requirements.txt``: ``PyYAML``.

**Comment by rhatdan** [comment]
@ashcrow PTAL at https://github.com/projectatomic/atomic/pull/538

**Comment by rhatdan** [comment]
Fixed in atomic-1.13
